### PR TITLE
Expiring soon runs of Overview page

### PR DESCRIPTION
**Pull request by MVarshini**
PBENCH - 730

Expiring soon runs

List the runs with server deletion date less than 20 days

### CODE REVIEW

File: dashboard/src/modules/components/OverviewComponent/common-component.jsx [hunk 1]
```diff
@@ -33,7 +40,7 @@ export const NoExpiringRuns = () => {
       <TextContent className="no-runs-wrapper">
         <Text component={TextVariants.h3}> You have no runs expiring soon</Text>
         <Text component={TextVariants.p}>
-          Runs that have expiration date within next 10days will appear here.
+          Runs that have expiration date within next 20 days will appear here.
```
Comment (dbutenhof) [inline dashboard/src/modules/components/OverviewComponent/common-component.jsx:43]: It'd be nice to use `{EXPIRATION_DAYS_LIMIT}` here rather than an inline number; if it changes, we don't need to remember to change it in both places.

(Also, at some point, we probably want to make this a user configuration setting rather than a hardcoded value; this change will at least make the dependency visible to code.)

File: dashboard/src/modules/components/TableComponent/index.jsx [hunk 2]
```diff
@@ -151,7 +147,7 @@ const TableWithFavorite = () => {
               redirect="auth"
             />
           )}
-          <div className="table-container" variant={PageSectionVariants.light}>
+          <div className="table-container">
```
Comment (dbutenhof) [inline dashboard/src/modules/components/TableComponent/index.jsx:150]: You've now put this fix into two separate PRs, but it's breaking others as well. I suspect it would make sense to pull this out into a separate PR that we can review and merge quickly!

**Review by dbutenhof — CHANGES_REQUESTED** [review]
This seems to be good. Looking forward to seeing it in operation! I have a few comments, including especially pulling the `variant` fix out into a separate PR.

File: dashboard/src/modules/components/TableComponent/index.jsx (hunk 2 above)
Comment (MVarshini) [reply dashboard/src/modules/components/TableComponent/index.jsx]: Moved it to a separate PR

File: dashboard/src/modules/components/OverviewComponent/common-component.jsx (hunk 1 above)
Comment (MVarshini) [reply dashboard/src/modules/components/OverviewComponent/common-component.jsx]: Updated!

File: dashboard/src/modules/components/OverviewComponent/common-component.jsx [hunk 3]
```diff
@@ -33,9 +41,10 @@ export const NoExpiringRuns = () => {
       <TextContent className="no-runs-wrapper">
         <Text component={TextVariants.h3}> You have no runs expiring soon</Text>
         <Text component={TextVariants.p}>
-          Runs that have expiration date within next 10days will appear here.
-          These runs will be automatically removed from the system if left
-          unacknowledged. <Button variant="link">Learn More.</Button>
+          Runs that have expiration date within next {EXPIRATION_DAYS_LIMIT}{" "}
```
Comment (dbutenhof) [inline dashboard/src/modules/components/OverviewComponent/common-component.jsx:44]: Is the forced space insert necessary here? That suggests JSX is somehow eating a real blank between the substitution text and "days"?? Huh.

**Review by dbutenhof — APPROVED** [review]

File: dashboard/src/modules/components/OverviewComponent/common-component.jsx (hunk 3 above)
Comment (webbnh) [reply dashboard/src/modules/components/OverviewComponent/common-component.jsx:44]: Yes, it seems odd that lines 44 and 46 need an explicit blank character when line 45 does not.  (Why does line 46 need it??)

File: dashboard/src/modules/components/OverviewComponent/common-component.jsx [hunk 4]
```diff
@@ -153,3 +162,26 @@ export const RenderPagination = (props) => {
...
+export const ProgressBar = (props) => {
+  const findVariant = () => {
+    if (props.percent >= 65) {
+      return null;
+    } else if (props.percent < 65 && props.percent >= 40) {
+      return ProgressVariant.warning;
+    } else {
+      return ProgressVariant.danger;
+    }
+  };
```
Comment (webbnh) [inline dashboard/src/modules/components/OverviewComponent/common-component.jsx:174]: At line 169, `props.percent < 65` will always be true, because if it were false then control would have entered the block at line 168 and would have never reached line 169.

Also, I wouldn't be opposed to using a compound ternary in spots like this:
```
  const findVariant = () => (
    props.percent >= 65
      ? null
      : props.percent >= 40
        ? ProgressVariant.warning
        : ProgressVariant.danger;
  );
```

**Review by webbnh — APPROVED** [review]
Looks good, but I have a small suggestion (and an addition to Dave's question...).

**Review by portante — APPROVED** [review]

File: dashboard/src/modules/components/OverviewComponent/common-component.jsx (hunk 3 above)
Comment (MVarshini) [reply dashboard/src/modules/components/OverviewComponent/common-component.jsx:44]: Actually, Line 46 does not require a blank character. But between the substitution text and "days" we need a blank character

File: dashboard/src/modules/components/OverviewComponent/common-component.jsx (hunk 4 above)
Comment (MVarshini) [reply dashboard/src/modules/components/OverviewComponent/common-component.jsx:174]: That's true. As we are not using this Progress Bar removed this component

**Review by dbutenhof — APPROVED** [review]

**Review by portante — APPROVED** [review]

File: dashboard/src/modules/components/OverviewComponent/common-component.jsx [hunk 5]
```diff
@@ -42,8 +36,8 @@ export const NoExpiringRuns = () => {
         <Text component={TextVariants.h3}> You have no runs expiring soon</Text>
         <Text component={TextVariants.p}>
           Runs that have expiration date within next {EXPIRATION_DAYS_LIMIT}{" "}
-          days will appear here. These runs will be automatically removed from
-          the system if left unacknowledged.{" "}
+          days will appear here. These runs will be automatically removed from{" "}
+          the system if left unacknowledged.
           <Button variant="link">Learn More.</Button>
         </Text>
```
Comment (webbnh) [inline dashboard/src/modules/components/OverviewComponent/common-component.jsx:42]: I hate to even ask, but....

When this text component is rendered, the text is not copied literally (with just the `{}` substitutions replaced) -- it is rewrapped (or "reflowed" or whatever the term is) and the line breaks and indentation are removed...right?  Are you saying that, e.g., without the `{" "}` at the end of line 39, the `from` there would be concatenated to the `the` from the start of the next line with no space between them?...and that something similar happens when the text from lines 38 and 39 are joined together?

**Review by webbnh — APPROVED** [review]
Looks good to me...but I have a question which is annoying me.

File: dashboard/src/modules/components/OverviewComponent/common-component.jsx (hunk 5 above)
Comment (webbnh) [reply dashboard/src/modules/components/OverviewComponent/common-component.jsx:42]: Some small experimentation suggests that the text is indeed re-flowed, with line breaks removed and sequences of multiple spaces replaced by a single space.  _However_, the `{}` expression at the end of a line does indeed get joined with the text from the next line with no intervening space (which strikes me as odd).  I suggest finessing this like:
```
        <Text component={TextVariants.p}>
          Runs that have expiration date within next {EXPIRATION_DAYS_LIMIT} days
          days will appear here. These runs will be automatically removed from
          the system if left unacknowledged.
          <Button variant="link">Learn More.</Button>
        </Text>
```
That shouldn't require any explicit spaces, and I don't think it makes the lines any longer (i.e., `{" "}` is five characters, just like `days` is if you count the preceding space 🙂).

Comment (webbnh) [reply dashboard/src/modules/components/OverviewComponent/common-component.jsx:42]: Or, we can just merge it as it is...your call, Varshini.

Comment (MVarshini) [reply dashboard/src/modules/components/OverviewComponent/common-component.jsx:42]: Let's merge for now. I'll check your suggestion later :grinning:

**Review by npalaska — APPROVED** [review]
